src/ripple/app/ledger/impl/SkipListAcquire.h

```diff
@@ -132,18 +146,18 @@ class SkipListAcquire final
         ScopedLockType& sl);
 
     /**
-     * Notify existing LedgerReplayTasks that this subtask is done
+     * Call the OnSkipListDataCB callbacks
      * @param sl  lock. this function must be called with the lock
      */
     void
-    notifyTasks(ScopedLockType& sl);
+    notify(ScopedLockType& sl);
 
     InboundLedgers& inboundLedgers_;
     LedgerReplayer& replayer_;
     std::uint32_t ledgerSeq_ = 0;
     std::unique_ptr<PeerSet> peerSet_;
-    std::vector<ripple::uint256> skipList_;
-    std::list<std::weak_ptr<LedgerReplayTask>> tasks_;
+    std::shared_ptr<std::vector<ripple::uint256> const> skipList_;
```

Comment:
Why did you wrap this in `std::shared_ptr`?

Reply:
Good question! I had to think and ended up wrapping more data. :-) Let's see if it makes sense. I wrapped the `ledgerSeq` together with the `skipList` as a const object, and changed how tasks get the `skipList` data. I think the `shared_ptr` is cheaper to copy (a skip list is 8KB), and the meaning of an empty `shared_ptr` is clear (in our case, "data is not ready") so it is easier to use. We currently limit the ledger range that can be replayed to 256, to have a simpler design. One future usage case needs larger range. It could be handled with multiple tasks, or we could improve the design of ledger replay and allow multiple `SkipListAcquire` per task. I prefer the latter case. I think the latest changes make that a bit simpler.